packages/@ember/-internals/runtime/lib/system/core_object.js

```diff
@@ -1120,4 +1173,21 @@ if (!HAS_NATIVE_SYMBOL) {
   });
 }
 
+// TODO: customize messages and add more debugging info like obj and mismatch information
+function implicitInjectionDeprecation(keyName, msg = null) {
+  deprecate(
+    msg,
+    false,
+    {
+      id: 'ember-object.implicit-injection',
+      until: '4.0.0',
+      url: 'https://',
+      for: 'ember-source',
+      since: {
+        enabled: '3.24.0',
+      },
+    }
+  );
+}
```

Comment:
I would generally prefer to just call `deprecate` multiple times, and not use this wrapper function 🤔